[Distributed Autograd] Make debugInfoMap from strings to ints

osalpekar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38416 [Distributed Autograd] Make debugInfoMap from strings to ints**

This diff primarily changes the `debugInfoMap` to map from strings to ints, instead of strings to strings. We were basically just converting these back to ints in Python so this avoid the extra conversions.

 `arc lint` also exposed tons of linting issues so fixing those here as well.

Differential Revision: [D21266522](https://our.internmc.facebook.com/intern/diff/D21266522/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21266522/)!

osalpekar committed:
[Distributed Autograd] Make debugInfoMap from strings to ints

This diff primarily changes the `debugInfoMap` to map from strings to ints, instead of strings to strings. We were basically just converting these back to ints in Python so this avoid the extra conversions.

 `arc lint` also exposed tons of linting issues so fixing those here as well.

Differential Revision: [D21266522](https://our.internmc.facebook.com/intern/diff/D21266522/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21266522/)!

[ghstack-poisoned]

mrshenli reviewed (approved):
This should be OK for now. But in the future, we might need to change this again into double or even back to string if we need to support more types.

osalpekar committed:
Update on "[Distributed Autograd] Make debugInfoMap from strings to ints"


This diff primarily changes the `debugInfoMap` to map from strings to ints, instead of strings to strings. We were basically just converting these back to ints in Python so this avoid the extra conversions.

 `arc lint` also exposed tons of linting issues so fixing those here as well.

Differential Revision: [D21266522](https://our.internmc.facebook.com/intern/diff/D21266522/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21266522/)!

[ghstack-poisoned]
